@ecomplus/storefront-components/src/js/TheProduct.js

```diff
@@ -492,14 +509,57 @@ export default {
 
   created () {
     if (this.product) {
-      this.body = this.product
+      this.body = { ...this.product }
       if (this.isSSR) {
         this.fetchProduct()
       }
     } else {
       this.fetchProduct()
     }
     this.isFavorite = checkFavorite(this.body._id || this.productId, this.ecomPassport)
+
+    // Just for number length check
+    const formatTime = (number) => {
+      return number.toString().length === 1 ? `0${number}` : number
+    }
+
+    // Retrieve promo timer
+    const getRemainingPromoTime = (c, t) => {
+      const distance = t - c;
+      // Uncomment days if you wish to show the full promo time
+      const days = Math.floor(distance / (1000 * 60 * 60 * 24))
+      const hours = Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60))
+      const minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60))
+      const seconds = Math.floor((distance % (1000 * 60)) / 1000)
+      return `${days > 0 ? `${formatTime(days)}:` : ''}${formatTime(hours)}:${formatTime(minutes)}:${formatTime(seconds)}`
+    }
+
+    // Retrieve timestamps
+    const currentTime = new Date().toISOString()
+    const saleEndTime = this.isOnSale
+      ? new Date(this.body.price_effective_date.end)
+      : null
+    const targetTime = saleEndTime
+      ? saleEndTime
+      : new Date(`${currentTime.split('T')[0]}T23:59:59`)
+
+    // Flags
+    const daysBetween = Math.floor((targetTime.getTime() - new Date().getTime()) / (1000 * 60 * 60 * 24))
+
```

Comment:
Tudo isso é inútil se `!this.isOnSale`, então tudo isso deveria estar dentro do `if (this.isOnSale) {`, certo?

Reply:
Resolvido